orchestration/src/main/java/com/sap/ai/sdk/orchestration/OrchestrationAiModel.java

```diff
@@ -114,4 +123,28 @@ public class OrchestrationAiModel {
   LLMModuleConfig createConfig() {
     return new LLMModuleConfig().modelName(name).modelParams(params).modelVersion(version);
   }
+
+  /**
+   * Subclass to allow for parameterized models.
+   *
+   * @param <T> The type of parameters for this model.
+   */
+  public static final class Parameterized<T extends OrchestrationAiModelParameters>
```

Comment:
why do we need this class?

Reply:
Assuming the [LLM providers expect individual sets of parameters](https://docs.litellm.ai/docs/completion/input#translated-openai-params) the additional `withParams( T )` would require a generic parameter type argument. As you can see VertexAI would not understand all your OpenAI arguments. To model this distinction a specific type argument (i.e. type hinting) is advantageous.